src/docfx/restore/Restore.cs

```diff
@@ -19,83 +19,77 @@ public static async Task Run(string docsetPath, CommandLineOptions options, Erro
                 var repository = Repository.Create(docsetPath);
                 Telemetry.SetRepository(repository?.Remote, repository?.Branch);
 
-                var restoredDocsets = new ConcurrentDictionary<string, Task<DependencyLockModel>>(PathUtility.PathComparer);
+                var restoredGitLock = new Dictionary<string, string>();
                 var localeToRestore = LocalizationUtility.GetLocale(repository?.Remote, repository?.Branch, options);
 
-                await RestoreDocset(docsetPath, rootRepository: repository);
+                var (errors, config) = ConfigLoader.TryLoad(docsetPath, options, localeToRestore, extend: false);
 
-                Task<DependencyLockModel> RestoreDocset(string docset, bool root = true, Repository rootRepository = null, DependencyLockModel dependencyLock = null)
-                {
-                    return restoredDocsets.GetOrAdd(docset + dependencyLock?.Commit, async k =>
-                    {
-                        var (errors, config) = ConfigLoader.TryLoad(docset, options, localeToRestore, extend: false);
-
-                        if (root)
-                        {
-                            errorLog.Configure(config);
-                        }
-                        errorLog.Write(errors);
-
-                        // no need to restore child docsets' loc repository
-                        return await RestoreOneDocset(
-                            docset,
-                            localeToRestore,
-                            config,
-                            (subDocset, subDependencyLock) => RestoreDocset(subDocset, root: false, dependencyLock: subDependencyLock),
-                            rootRepository,
-                            dependencyLock,
-                            root);
-                    });
-                }
-            }
+                var (fallbackErrors, fallbackConfig, restoreFallbackResult) = RestoreFallbackRepo(docsetPath, config, repository, localeToRestore, options);
+                errors.AddRange(fallbackErrors);
+
+                config = fallbackConfig ?? config;
```

Comment:
One suggestion:
the returned `fallbackConfig` would be `null` if current repo has config. 
But `config = fallbackConfig ?? config;` is more like saying `fallbackConfig` has higher priority.

Can we get rid of this misleading by making `RestoreFallbackRepo` always return a `fallbackConfig`, and uses
`config = config ?? fallbackConfig;`?

Reply:
Good suggestion, but I think for transparent purpose, we should move this logic out of RestoreFallbackRepo

    the returned fallbackConfig would be null if current repo has config